codalab/rest/bundle_actions.py

```diff
@@ -17,20 +17,36 @@ def create_bundle_actions():
     actions = BundleActionSchema(strict=True, many=True).load(request.json).data
 
     check_bundles_have_all_permission(local.model, request.user, [a['uuid'] for a in actions])
-
     for action in actions:
         bundle = local.model.get_bundle(action['uuid'])
-        if bundle.state not in [State.RUNNING, State.PREPARING]:
-            raise UsageError('Cannot execute this action on a bundle that is not running.')
+        if bundle.state not in [
+            State.CREATED,
+            State.UPLOADING,
+            State.STAGED,
+            State.PREPARING,
+            State.RUNNING,
+            State.FINALIZING,
+        ]:
+            raise UsageError(
```

Comment:
At this point, isn't better to define this in terms of terminal states (ready, failed)?

Reply:
Sure. I was debating if I should list all the states here or just using terminal states. Updating it.